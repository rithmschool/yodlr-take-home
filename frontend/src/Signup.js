import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { addUser } from './reducer/users/action'
import { useHistory } from 'react-router-dom';


const Signup = () => {
  const error = useSelector(state => state.error);
  const INIT_FORM_STATE = {
    "firstName": "",
    "lastName": "",
    "email": "",
  }
  const [formData, setFormData] = useState(INIT_FORM_STATE);
  const [formError, setFormError] = useState("");
  const dispatch = useDispatch();
  const history = useHistory();

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData(f => ({ ...f, [name]: value }));
  }

  const handleSubmit = (evt) => {
    evt.preventDefault();
    try {
      dispatch(addUser({ ...formData, state: "pending" }));
      setFormData(INIT_FORM_STATE);
      history.push('/');
    } catch (e) {
      console.error(e);
      setFormError(error);
    }
  }


  return (
    <div className="row justify-content-center mt-5">
      <div className="card col-md-6">
        <form className="card-body text-left" onSubmit={handleSubmit}>
          <div className="form-row">
            <div className="form-group col-md-6">
              <label htmlFor="intput-first-name">First Name</label>
              <input type="text" className="form-control" id="intput-first-name"
                name="firstName" value={formData.firstName} onChange={handleChange} />
            </div>
            <div className="form-group col-md-6">
              <label htmlFor="intput-last-name">Last Name</label>
              <input type="text" className="form-control" id="intput-last-name"
                name="lastName" value={formData.lastName} onChange={handleChange} />
            </div>
          </div>
          <div className="form-group">
            <label htmlFor="input-email">Email</label>
            <input type="text" className="form-control" id="input-email"
              name="email" value={formData.email} onChange={handleChange} />
          </div>
          <button type="submit" className="btn btn-primary btn-block"
            disabled={Object.values(formData).some(field => field.length === 0)}>
            Sign up
          </button>
          {formError.length !== 0 &&
            <div className="alert alert-danger" role="alert">
              {formError}
            </div>
          }
        </form>
      </div>
    </div>
  )
}


export default Signup;