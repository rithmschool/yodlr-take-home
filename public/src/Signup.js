import React, { useState } from 'react';
import { Form, FormGroup, Button, Input } from 'reactstrap';

function Signup ({ signup }) {
    const initialState = {firstName: "", lastName: "", email: ""}
    const [formData, setFormData] = useState(initialState)

    // handle user form input before submit
    function handleChange (evt) {
        const { name, value } = evt.target;
        setFormData(fData => ({
          ...fData,
          [name]: value
        }));
    }
    
    // handle form submission
    async function handleSubmit (evt) {
        evt.preventDefault();
        try{
        await signup(formData);
        setFormData(initialState);
        } catch (err) {
            return( <p>Bad login, please try again.</p>)
        }
    }


    // display signup form
    return (
        <div>
            <h2>Sign up for updates about our company!</h2>
            <Form onSubmit={handleSubmit}>
                <FormGroup>
                    <Input type="text" name="firstName" placeholder="First name" onChange={handleChange} required></Input>
                </FormGroup>
                <FormGroup>
                    <Input type="text" name="lastName" placeholder="Last name" onChange={handleChange} required></Input>
                </FormGroup>
                <FormGroup>
                    <Input type="text" name="email" placeholder="Email" onChange={handleChange} required></Input>
                </FormGroup>
                <Button>Sign up</Button>
            </Form>
        </div>
    )
}

export default Signup;