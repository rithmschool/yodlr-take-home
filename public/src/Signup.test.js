import React from "react";
import { render } from "@testing-library/react";
import Signup from './Signup';

it("renders without crashing", function() {
  render(<Signup />);
});

it("matches snapshot", function() {
  const {asFragment} = render(<Signup />);
  expect(asFragment()).toMatchSnapshot();
});