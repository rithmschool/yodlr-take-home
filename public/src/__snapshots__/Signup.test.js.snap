// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Sign up for updates about our company!
    </h2>
    <form
      class=""
    >
      <div
        class="form-group"
      >
        <input
          class="form-control"
          name="firstName"
          placeholder="First name"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-group"
      >
        <input
          class="form-control"
          name="lastName"
          placeholder="Last name"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-group"
      >
        <input
          class="form-control"
          name="email"
          placeholder="Email"
          required=""
          type="text"
        />
      </div>
      <button
        class="btn btn-secondary"
      >
        Sign up
      </button>
    </form>
  </div>
</DocumentFragment>
`;
